CMakeLists.txt

```diff
@@ -10,10 +10,33 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH ON)
 
-find_package(fmt REQUIRED)
-find_package(spdlog REQUIRED)
+include(FetchContent)
+
+FetchContent_Declare(
+  fmt
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG        10.2.1
+  GIT_SHALLOW    TRUE
+)
+FetchContent_MakeAvailable(fmt)
+
+FetchContent_Declare(
+  spdlog
+  GIT_REPOSITORY https://github.com/gabime/spdlog.git
+  GIT_TAG        v1.13.0
+  GIT_SHALLOW    TRUE
```

Comment:
You could also try adding

```cmake
CONFIGURE_COMMAND   "-DSPDLOG_FMT_EXTERNAL=On"
```
as one of the arguments to `spdlog`'s `FetchContent_declare`, but I don't know if that will work.

Reply:
Maybe this is better:

```diff
- CONFIGURE_COMMAND "-DSPDLOG_FMT_EXTERNAL=On"
+ CMAKE_ARGS "-DSPDLOG_FMT_EXTERNAL=On"
```